test/test_browser.py

```diff
@@ -4690,6 +4690,7 @@ def test_webgl_simple_extensions(self, simple_enable_extensions, webgl_version):
   @parameterized({
     'default': ([],),
     'closure': (['-sASSERTIONS', '--closure=1'],),
+    'closure_advanced': (['-sASSERTIONS', '--closure=1', '-O3'],),
```

Comment:
I'm not sure we need both `-O0` and `-O3` here.

Reply:
There are a few other tests that test closure with and without O3 so I was guessing there must be some expected difference:
https://github.com/emscripten-core/emscripten/blob/d278e2d0a0b9a86a880dfa57b3794721e0a3ccf4/test/test_other.py#L9650-L9656
Though I guess that could just be trying to change the optimization level of the Wasm.

Regardless in practice it does seem to make a difference. Though as mentioned it somehow seemed nondeterministic.

Reply:
Sounds good to me to land this as is then.

The nondeterministic part is certainly very worrying though.  If you can share a concrete example of that I would love to to see it.. there should not be any non-determinism in the toolchain clearly.